File: api/BeSureApi/Models/BusinessFunction.cs
namespace BeSureApi.Models
{
    public class BusinessFunctionList
    {
        public int Id { get; set; }
        public string BusinessFunctionCode { get; set; }
        public string BusinessFunctionName { get; set; }
        public string BusinessModuleName { get; set; }
        public string? Description { get; set; }
        public bool IsActive { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public string? UpdatedBy { get; set; }
        public DateTime? UpdatedOn { get; set; }
    }
}

File: ui/src/types/businessFunction.ts
import { Decoder, array, boolean, nullable, number, object, string } from "decoders";

export interface BusinessFunctionDetails {
    Id: number;
    BusinessFunctionCode: string;
    BusinessFunctionName: string;
    BusinessModuleName:string;
    IsActive: boolean;
    CreatedBy: string;
    CreatedOn: string;
    UpdatedBy: null | string;
    UpdatedOn: null | string;
  }

  export const businessFunctionDetailsDecoder : Decoder<BusinessFunctionDetails> = object({
    Id: number,
    BusinessFunctionCode: string,
    BusinessFunctionName: string,
    BusinessModuleName:string,
    IsActive: boolean,
    CreatedBy: string,
    CreatedOn: string,
    UpdatedBy: nullable(string),
    UpdatedOn: nullable(string)
  });  
  
  export interface BusinessFunctionList{
    BusinessFunction:BusinessFunctionDetails[],
    TotalRows: number;
    PerPage:number;
  }

  export const businessFunctionListDecoder: Decoder<BusinessFunctionList> = object({
    BusinessFunction: array(businessFunctionDetailsDecoder),
    TotalRows: number,
    PerPage:number
  });